model_signing/signature/verifying.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This package provides the functionality to verify signed models."""
+import abc
+
+from dataclasses import dataclass
+from sigstore_protobuf_specs.dev.sigstore.bundle import v1 as bundle_pb
+
+
+@dataclass(frozen=True)
+class VerificationResult:
+    passed: bool
+    information: str = ''
+
+
+class Verifier(abc.ABC):
+    """Verifier is the abstract base class for all verifying methods."""
+
+    @abc.abstractmethod
+    def verify(self, bundle: bundle_pb.Bundle) -> VerificationResult:
```

Comment:
What if we make this raise an error if verification fails and return `None` (don't return anything) otherwise?

Reply:
I felt it's cleaner to not raise an error and return a data structure describing the outcome but I don't have strong feelings about it.

Reply:
Changed it to exception based